<?xml version="1.0" encoding="utf-8"?>
<robot name="dofbot">

    <link name="base_link"/>
    <link name="dofbot_base_link">
        <visual>
            <origin 
                xyz="0 0 0"
                rpy="0 0 0"/>
            <geometry> <cylinder radius="0.05" length="0.005"/> </geometry>
            <material name=""> <color rgba="0 0 0 0.5"/> </material>
        </visual>
    </link>
    <joint name="fixed_base_joint" type="fixed">
        <parent link="base_link"/>
        <child link="dofbot_base_link"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>

    <link name="link1">
        <visual>
            <origin
                xyz="0 0 0.02"
                rpy="0 0 0"/>
            <geometry> <cylinder radius="0.0225" length="0.04"/> </geometry>
            <material name="">
                <color rgba="0 0.7 0 0.3"/>
            </material>
        </visual>
        <collision>
            <origin
                xyz="0 0 0.02"
                rpy="0 0 0"/>
            <geometry> <cylinder radius="0.0225" length="0.04"/> </geometry>
        </collision>
    </link>
    <joint name="joint1" type="fixed">
        <parent link="dofbot_base_link"/>
        <child link="link1"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>

    <link name="link2">
       <visual>
            <origin xyz="0 0 0.05" rpy="0 0 0"/>
            <geometry>
                <box size="0.03 0.03 0.10"/>
            </geometry>
            <material name="">
                <color rgba="0.7 0.7 0.7 0.5"/>
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0.05" rpy="0 0 0"/>
            <geometry>
                <box size="0.03 0.03 0.10"/>
            </geometry>
        </collision>
    </link>
    <joint name="joint2" type="revolute">
        <parent link="link1"/>
        <child link="link2"/>
        <!-- link1의 위쪽 중심부에서 2.5cm 겹치도록 아래쪽으로 이동 -->
        <origin xyz="0 0 0.015" rpy="0 0 0"/>
        <!-- x축을 중심으로 회전 -->
        <axis xyz="1 0 0"/>
        <limit
            lower="-1.5708"
            upper="1.5708"
            effort="100"
            velocity="1"/>
    </joint>

    <link name="link3">
        <visual>
            <origin xyz="0 0 0.029" rpy="0 0 0"/>  <!-- 높이의 절반 -->
            <geometry>
                <box size="0.02 0.042 0.058"/>
            </geometry>
            <material name="">
                <color rgba="0 0.7 0 1"/>
            </material>
        </visual>
        <collision>
            <origin xyz="0 0 0.029" rpy="0 0 0"/>
            <geometry>
                <box size="0.02 0.042 0.058"/>
            </geometry>
        </collision>
    </link>
    <joint name="joint3" type="revolute">
        <parent link="link2"/>
        <child link="link3"/>
        <origin xyz="0 0 0.08" rpy="0 0 0"/>  <!-- link2의 꼭대기에서 아래로 2cm -->
        <axis xyz="0 1 0"/>  <!-- Y축 기준 회전 -->
        <limit
            lower="-1.5708"
            upper="1.5708"
            effort="100"
            velocity="1"/>
    </joint>
</robot>